E1_TFHub_Sample_Deploy/export_test.py

```diff
@@ -0,0 +1,55 @@
+import tensorflow as tf
+import numpy as np
+import tensorflow_datasets as tfds
+import tensorflow_hub as hub
+import unittest
+from absl import app, flags
+
+flags.DEFINE_string(
+    "exported_path",
+    "/tmp/tfhub_modules/mnist/digits/1",
+    "Path to exported Module")
+flags.DEFINE_string("data_dir", None, "Path to TFDS Dataset Directory")
+flags.DEFINE_boolean("check_accuracy", False, "Checks the accuracy of a model")
+flags.DEFINE_integer("batch_size", 32, "Batch size for checking accuracy")
+flags.DEFINE_integer(
+    "buffer_size",
+    1000,
+    "Buffer size used for Data Shuffling")
+FLAGS = flags.FLAGS
+
+
+class TFHubMNISTTest(tf.test.TestCase):
+  def setUp(self):
+    kwargs = {}
+    if FLAGS.data_dir:
+      kwargs = {"data_dir": FLAGS.data_dir}
+    self.dataset = tfds.load("mnist",
+                             split="test",
+                             **kwargs).shuffle(FLAGS.buffer_size,
+                                               reshuffle_each_iteration=True).batch(FLAGS.batch_size)
+    self.batch = FLAGS.batch_size
+    self.accuracy_flag = FLAGS.check_accuracy
+    self.model = hub.load(FLAGS.exported_path)
+
+  def checkModelLoaded(self):
+    output_ = self.model.call(
+        tf.zeros([self.batch, 28, 28, 1], dtype=tf.uint8).numpy())
+    self.assertEqual(output_.shape, [self.batch, 10])
+
+  def checkModelWorking(self):
+    sample = next(iter(self.dataset.take(1)))
+    prediction = self.model.call(sample['image'])
+    self.assertEqual(tf.argmax(prediction).numpy(), sample['label'])
+
+  def checkAccuracy(self):
+    accuracy = 0
+    for data in self.dataset:
+      preds = self.model.call(data['image'])
+      preds = tf.argmax(preds, axis=1).numpy()
+      accuracy += sum(preds == data['label'].numpy()) / self.batch
+    print("Accuracy: %.2f" % (accuracy * 100))
```

Comment:
Remove print statements from tests.

Reply:
Well, this function actually tests the accruacy of the model. So, where should I log the accuracy? I know, tests are not meant for this purpose, but I'm out of ideas for tests. 😅
Can you suggest some?

Reply:
We shouldn't be loading an entire dataset in unit tests. Unit tests are usually run as a sanity check anyway.
Ideally, we should verify the outputs (via asserts) for one or two sample inputs (just put raw images in a ['testdata' folder](https://github.com/tensorflow/tensorflow/tree/master/tensorflow/core/lib/jpeg) if need be), and boundary cases such as all-zeros, invalid data, etc. There should be one test per case, like [this](https://github.com/tensorflow/hub/blob/master/examples/text_embeddings/export_test.py#L86).

Reply:
Done :)